### ISSUE TITLE
Double Tap with Apple Pencil

### ISSUE THREAD

**Issue by LePenseur13** [zsviczian/obsidian-excalidraw-plugin#729]
**Is your feature request related to a problem? Please describe.**
As an Ipad user I’m used to use the double tap feature on my apple pencil, which speeds up my workflow.
Excalidraw is great, however the lack of this feature is the reason, why I quite often use another drawing app and the later import it.

**Describe the solution you'd like**
use the double tap to switch between drawing and eraser
Or even better: inherit the apple setting regarding how the double tap should work, which is either switch between drawing and eraser or betweeen drawing and last used tool.

**Describe alternatives you've considered**
Using another app and importing it. E.g. Goodnotes. Drawback: the whole drawing is then one object and the different seperate lines are not recognized. Hence if one could import from Goodnotes in a way that the seperate lines and objects would be recognized, this would be great as well.

**Additional context**
To implement the Apple Pencil double tap feature, would be a huge improvement for IPad users.
It increases usability, workflow and speed a lot.
especially if you are used to it.

With this feature Excalidraw would replace other apps like Good notes, Notability, etc.

**Comment by LePenseur13** [comment]
My own research:
There should be ways one can access native features in electron?

I found these posts: 
- https://bountify.co/electron-call-macos-requestreview-api (Executing native code in Electron app: with an example)
- https://stackoverflow.com/questions/34670514/can-i-use-osx-native-apis-in-electron

Couldn’t in that direction a solution found?
I’m not an expert here, so sorry if I missed something important.

**Comment by zsviczian** [comment]
I agree that this would be great... unfortunately I haven't found a way to make it happen.
- The Obsidian mobile app is not running on Electron, but Safari on iPad and Web View on Android (typically Chromium).
- To the best of my knowledge (online research) Apple has not made the pen tap action available to Safari via javascript.
- The underlying issue is a lack of standardization between styluses. Next to the Apple Pencil, there are a dozen others... all with different configurations. Creating code that caters to all is just simply not maintainable.
- On my desktop, I have configured the button of my Wacom pen to simulate the "E" click. "E" toggles between the eraser and the active tool. This works great on the destkop. It would be great if Apple allowed such a configuration... but I guess that is never going to happen.

If someone points me to a javascript library that works on Safari and can access the pencil button, I am more than happy to take a look.

**Comment by LePenseur13** [comment]
I understand. That’s too bad.

**Comment by LePenseur13** [comment]
This is the only thing I found so far. https://www.npmjs.com/package/capacitor-apple-pencil/v/0.1.0
However it is no longer maintained.

**Comment by LunaVoid** [comment]
I looked a bit more into thisIt seems like you do this with a WKwebview component: https://developer.apple.com/documentation/webkit/wkwebview and https://stackoverflow.com/questions/49142047/detect-whether-touch-was-apple-pencil-or-finger-in-webview-react-native
But from what I can make out this seems like a pretty substantial amount of work for a simple gesture. So it makes sense for this to be categorized as such. Theoretically, I feel like using a WKwebview when this plugin is used on and ipad might increase performance as well but I really am not an expert on that.

**Comment by zsviczian** [comment]
The webview component hosting Obsidian is part of the Obsidian build, i.e. not something I could do within the plugin. Now, the interesting question is if Obsidian is already using this already using wkwebview.... and if not, if they are willing to to consider.

**Comment by zsviczian** [comment]
Consolidating pen / stylus / handwriting / drawing tablet support discussion into #2773.

Closing this as a duplicate so related requests stay centralized. As noted in #2773, native vendor-specific SDK features such as Apple Pencil double tap are unlikely to be supported unless Obsidian exposes them to plugins in a sustainable way.
